### ISSUE TITLE
Shopify-buy unable to retrieve several parameteres 

### ISSUE THREAD

**Issue by RamJogal** [Shopify/js-buy-sdk#949]
Hi team
, team. When I try to access Shopify products using the npm package Shopify -buy, I get the error shown above.
Framework 
React 

Shopify-buy Version

2.19.0
```
 {
    "message": "Selections can't be made on scalars (field 'price' returns Money but has selections [amount, currencyCode])",
    "locations": [
        {
            "line": 1,
            "column": 56
        }
    ],
    "path": [
        "fragment VariantFragment",
        "price"
    ],
    "extensions": {
        "code": "selectionMismatch",
        "nodeName": "field 'price'",
        "typeName": "Money"
    }
}
```

**Comment by JeffJassky** [comment]
This is also happening to me - all of a sudden out of nowhere. Any chance you found a work around?

**Comment by RamJogal** [comment]
Yes This was Beacuse While Using the SDK i wasn't providing the API version as " " in the arguments so of which many parameters were missing, to solve the above issue i added some value to API version  in  the argument now every thing is working fine

**Comment by JeffJassky** [comment]
This is also happening to me #960 and others #959

**Comment by JeffJassky** [comment]
> Yes This was Beacuse While Using the SDK i wasn't providing the API version as " " in the arguments so of which many parameters were missing, to solve the above issue i added some value to API version in the argument now every thing is working fine

I can confirm that graphql request URLs are using the default apiVersion that is defined in the library.

However, which apiVersion are you using? And where are you defining it?

I've attempted a few alternatives like this with no success:
```
export default Client.buildClient({
  domain: '*****************',
  storefrontAccessToken: '******************************',
  apiVersion: '2023-07'
});
```

**Comment by RamJogal** [comment]
Hi Jeff My bad Earlier I kept the API Version as " " in arguments after that I tried that with some values everything was working fine so after that i removed the API version argument from the arguments 
this.client = Client.buildClient({
      domain,
      storefrontAccessToken,
    });
